src/testcafe-runner.js

```diff
@@ -96,33 +97,37 @@ async function run (runCfgPath, suiteName) {
     });
 
     let endTime = new Date().toISOString();
+    passed = results === 0;
 
-    // Retain the assets now
-    if (process.env.SAUCE_USERNAME && process.env.SAUCE_ACCESS_KEY && !process.env.SAUCE_VM) {
-      console.log(`Reporting assets in '${assetsPath}' to Sauce Labs`);
-      await sauceReporter({
-        browserName,
-        assetsPath,
-        results,
-        assets: [
-          path.join(assetsPath, 'report.xml'),
-          path.join(assetsPath, 'report.json'),
-          path.join(assetsPath, 'video.mp4'),
-          path.join(assetsPath, 'console.log'),
-        ],
-        startTime,
-        endTime,
-      });
-    } else if (!process.env.SAUCE_VM) {
+    if (process.env.SAUCE_VM) {
+      return passed;
+    }
+    if (!process.env.SAUCE_USERNAME && !process.env.SAUCE_ACCESS_KEY) {
       console.log('Skipping asset uploads! Remember to setup your SAUCE_USERNAME/SAUCE_ACCESS_KEY');
+      return passed;
     }
+
+    console.log(`Reporting assets in '${assetsPath}' to Sauce Labs`);
+    await sauceReporter({
+      browserName,
+      assetsPath,
+      results,
+      assets: [
+        path.join(assetsPath, 'report.xml'),
+        path.join(assetsPath, 'report.json'),
+        path.join(assetsPath, 'video.mp4'),
+        path.join(assetsPath, 'console.log'),
+      ],
+      startTime,
+      endTime,
+    });
+
     passed = results === 0;
   } catch (e) {
     console.error(`Could not complete test. Reason '${e.message}'`);
-    passed = false;
```

Comment:
This is not required and may create what we are trying to avoid.

It's declared on L7 and being `undefined`: https://github.com/saucelabs/sauce-testcafe-runner/blob/fix-error-docker-mode/src/testcafe-runner.js#L7 

Forcing it to `false` at this specific place will meaning `failure` even if tests passed and it goes in the catch block due to the reporter. Value can be set to `false` at beginning, but it will keep the same behaviour.

This run part should be refactored, because this is complex to understand how things are going.

Reply:
But then the error is going be odd, no?
```
     console.error(`Could not complete test. Reason '${e.message}'`);
```


The call to `sauceReporter` should simply never return an error, or have its own catch-all block, so it doesn't affect the outer one 🤔

Reply:
@alexplischke I splitted the big `run` function into 3 parts:
- Config preparation
- TestCafe run
- Sauce Labs reporting

Each has its own catch block.